pkg/skaffold/debug/transform.go

```diff
@@ -311,3 +311,48 @@ func describe(obj runtime.Object) (group, version, kind, description string) {
 	}
 	return
 }
+
+// exposePort adds a `ContainerPort` instance or amends an existing entry with the same port.
+func exposePort(entries []v1.ContainerPort, portName string, port int32) []v1.ContainerPort {
+	found := false
+	for i := 0; i < len(entries); {
+		switch {
+		case entries[i].Name == portName:
+			// Ports and names must be unique so rewrite an existing entry if found
+			entries[i].Name = portName
+			entries[i].ContainerPort = port
```

Comment:
can we add a log warning here when there is a mismatch between the port and the existing ContainerPort? It would be useful to see that in case it happens.